python/triton/runtime/jit.py

```diff
@@ -640,11 +604,11 @@ def run(self, *args, grid, warmup, **kwargs):
             sigvals = sig_and_spec[:len(sigkeys)]
             signature = {k: ('*i8' if (v == 'none') else v) for (k, v) in zip(sigkeys, sigvals)}
 
-            configs = (self._get_config(*bound_vals), )
+            configs = (backend.get_attrs_descriptor(self.params, bound_vals), )
             constants = {
                 p.name: v
                 for (v, p) in zip(bound_vals, self.params)
-                if p.is_constexpr or p.num in configs[0].equal_to_1 or v is None
+                if p.is_constexpr or p.num in configs[0]["tt.equal_to_1"] or v is None
```

Comment:
This is hardcoding some properties. Do you see needs to allow each backend to inject something here? I guess it's hard given this is touching compiler-runtime interface.. Maybe we should be clear in the requirements for backend-specific properties at least.

Reply:
If I understand your comment, you are saying that a backend might not implement a "tt.equal_to_1" property. If this is the case, this will error. So either a) we impose the backend's AttrsDescriptor to have a "tt.equal_to_1" property, or b) we need a way to pass `p` or `p.num` to AttrsDescriptor to let the class tell us if `p` is a constant.

My idea was a bit to go with a). I implemented a `__get_item__` function in the base class that returns None if "tt.equal_to_1" is not there. So if the backend does not implement "tt.equal_to_1" this test will simply return `False` 

 To go with b) direction,I can add a method `is_constant(param)`. The default implementation will be to look a the "tt.equal_to_1" property.

Reply:
I'm actually asking the case where some additional backend specific properties may want to do something like `equal_to_1` is doing, which requires touching the interface when calling the kernel. I guess we can worry this a bit later when the needs arise and cap the backend-specific properties to not support such cases for now.